proteuslib/unit_models/nanofiltration_0D.py

```diff
@@ -197,46 +303,210 @@ def build(self):
+
+        self.avg_conc_mass_io_phase_comp = Var(
+            self.flowsheet().config.time,
+            self.io_list,
+            self.config.property_package.phase_list,
+            solute_set,
+            initialize=10,
+            bounds=(1e-3, 2e3),
+            units=pyunits.kg * pyunits.m ** -3,
+            doc="Average mass concentration at inlet and outlet")
+
+        self.area = Var(
+            initialize=10,
+            bounds=(1e-1, 1e3),
+            domain=NonNegativeReals,
+            units=units_meta('length') ** 2,
+            doc='Membrane area')
+
+        def recovery_mass_phase_comp_initialize(b, t, p, j):
+            if j in solvent_set:
+                return 0.1
+            elif j in solute_set:
+                return 0.01
+
+        def recovery_mass_phase_comp_bounds(b, t, p, j):
+            ub = 1 - 1e-6
+            if j in solvent_set:
+                lb = 1e-2
+            elif j in solute_set:
+                lb = 1e-5
+            return lb, ub
+
+        self.recovery_mass_phase_comp = Var(
             self.flowsheet().config.time,
             self.config.property_package.phase_list,
             self.config.property_package.component_list,
-            initialize=1e-3,
-            bounds=(1e-12, 1e6),
-            units=units_meta('mass')*units_meta('length')**-2*units_meta('time')**-1,
-            doc='Flux at feed outlet')
-        self.avg_conc_mass_phase_comp_in = Var(
+            initialize=recovery_mass_phase_comp_initialize,
+            bounds=recovery_mass_phase_comp_bounds,
+            units=pyunits.dimensionless,
+            doc='Mass-based component recovery')
+
+        self.recovery_vol_phase = Var(
             self.flowsheet().config.time,
             self.config.property_package.phase_list,
-            self.solute_list,
-            initialize=1e-3,
-            bounds=(1e-8, 1e6),
-            domain=NonNegativeReals,
-            units=units_meta('mass')*units_meta('length')**-3,
-            doc='Average solute concentration at feed inlet')
-        self.avg_conc_mass_phase_comp_out = Var(
+            initialize=0.1,
+            bounds=(1e-2, 1 - 1e-6),
+            units=pyunits.dimensionless,
+            doc='Volumetric-based recovery')
+
+        self.rejection_phase_comp = Var(
             self.flowsheet().config.time,
             self.config.property_package.phase_list,
-            self.solute_list,
-            initialize=1e-3,
-            bounds=(1e-8, 1e6),
-            domain=NonNegativeReals,
-            units=units_meta('mass')*units_meta('length')**-3,
-            doc='Average solute concentration at feed outlet')
-        self.area = Var(
-            initialize=1,
-            bounds=(1e-8, 1e6),
-            domain=NonNegativeReals,
-            units=units_meta('length') ** 2,
-            doc='Membrane area')
+            solute_set,
+            initialize=0.9,
+            bounds=(1e-2, 1 - 1e-6),
```

Comment:
Same as previous comment.

Reply:
The same idea applies here as in my response to the previous comment, except the variable in this case is solute rejection.